src/main/java/it/gov/pagopa/receipt/pdf/generator/utils/Aes256Utils.java

```diff
@@ -0,0 +1,89 @@
+package it.gov.pagopa.receipt.pdf.generator.utils;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.security.spec.KeySpec;
+import java.util.Base64;
+
+public class Aes256Utils {
+
+    private static final String AES_SECRET_KEY = System.getenv().getOrDefault("AES_SECRET_KEY", "");
+    private static final String AES_SALT = System.getenv().getOrDefault("AES_SALT", "");
+    private static final int KEY_LENGTH = 256;
+    private static final int ITERATION_COUNT = 65536;
+    public static final String PBKDF_2_WITH_HMAC_SHA_256 = "PBKDF2WithHmacSHA256";
+    public static final String AES_CBC_PKCS_5_PADDING = "AES/CBC/PKCS5Padding";
+
+    public static final String ALGORITHM = "AES";
+
+
+    /**
+     * Hide from public usage.
+     */
+    private Aes256Utils() {
+    }
+
+    public static String encrypt(String strToEncrypt) {
+
+        try {
+
+            SecureRandom secureRandom = new SecureRandom();
+            byte[] iv = new byte[16];
+            secureRandom.nextBytes(iv);
+            IvParameterSpec ivspec = new IvParameterSpec(iv);
+
+            SecretKeyFactory factory = SecretKeyFactory.getInstance(PBKDF_2_WITH_HMAC_SHA_256);
+            KeySpec spec = new PBEKeySpec(AES_SECRET_KEY.toCharArray(), AES_SALT.getBytes(), ITERATION_COUNT, KEY_LENGTH);
+            SecretKey tmp = factory.generateSecret(spec);
+            SecretKeySpec secretKeySpec = new SecretKeySpec(tmp.getEncoded(), ALGORITHM);
+
+            Cipher cipher = Cipher.getInstance(AES_CBC_PKCS_5_PADDING);
```

Comment:
Padding vulnerability rule java:S5542 ignored because encryption/decryption is used inside application workflow and is impossible to access it from the outside